lib/features/search/search.js

```diff
@@ -73,23 +82,29 @@ function matchItem(item, words, keys) {
     matchedWords,
     tokens
   } = keys.reduce((result, key) => {
-    const string = item[ key ];
+    let itemValue = item[ key ];
 
-    const {
-      tokens,
-      matchedWords
-    } = matchString(string, words);
+    if (!isArray(itemValue)) {
+      itemValue = [ itemValue ];
+    }
 
-    return {
-      tokens: {
-        ...result.tokens,
-        [ key ]: tokens,
-      },
-      matchedWords: {
-        ...result.matchedWords,
-        ...matchedWords
-      }
-    };
+    return itemValue.reduce((result, itemString, i) => {
+      const {
+        tokens,
+        matchedWords
+      } = matchString(itemString, words);
+
+      return {
+        tokens: {
+          ...result.tokens,
+          [ key + i]: (result.tokens[ key + i] || []).concat(tokens),
```

Comment:
I wonder if there is a way to "fix this", so that existing tokens are returned, i.e. we could concatenate the tokens (`tokens[key]` is an array when `item[ key ]` is an array).

Background: These tokens are, already today, consumed via the [search pad](https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/search-pad/SearchPad.js#L507), and potentially other places, and a these places can use the tokens to render matches, rely on the format. Your proposal breaks the search, I think.